### PR TITLE
[FLINK-24816][rabbitmq] Bump amqp-client to v5.13.1

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Update RabbitMQ Java client to the latest available version

## Brief change log

* Updated com.rabbitmq:amqp-client from 5.9.0 to 5.13.1

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@flinkbot run azure

File: flink-connectors/flink-connector-rabbitmq/pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@ under the License.
 
 	<!-- Allow users to pass custom connector versions -->
 	<properties>
-		<rabbitmq.version>5.9.0</rabbitmq.version>
```
Comment (fapaul) [inline flink-connectors/flink-connector-rabbitmq/pom.xml:40]: Should we also update our docker image `DockerImageVersions#RABBITMQ`?

**Review by fapaul — APPROVED** [review]
LGTM % Azure is also happy

**Comment by MartijnVisser** [comment]
@AHeise Please merge

**Pull request merged**